In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('loan_data.csv')

In [3]:
df

Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
1    LP001003    Male     Yes          1      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
..        ...     ...     ...        ...           ...           ...   
609  LP002978  Female      No          0      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         NaN             360.0   
1               4583             1508.0       128.0             360.0   
2               3000                0.0        66.0             360.0   
3               2583             2358.0       120.0             360.0   
4               6000                0.0       141.0             360.0   
..               ...                ...         ...               ...   
609             2900                0.0        71.0             360.0   
610             4106                0.0        40.0             180.0   
611             8072              240.0       253.0             360.0   
612             7583                0.0       187.0             360.0   
613             4583                0.0       133.0             360.0   

     Credit_History Property_Area Loan_Status  
0               1.0         Urban           Y  
1               1.0         Rural           N  
2               1.0         Urban           Y  
3               1.0         Urban           Y  
4               1.0         Urban           Y  
..              ...           ...         ...  
609             1.0         Rural           Y  
610             1.0         Rural           Y  
611             1.0         Urban           Y  
612             1.0         Urban           Y  
613             0.0     Semiurban           N  

[614 rows x 13 columns]

In [9]:
# which has the highest loan amount based on gender
df[df['Gender']=='Female']['LoanAmount'].max()

600.0

In [10]:
df[df['Gender']=='Male']['LoanAmount'].max()

650.0

### The highest loan amount by a male was 650 and 600 by a female

In [15]:
df[df['Gender']=='Female']['LoanAmount'].mean()

126.69724770642202

In [16]:
df[df['Gender']=='Male']['LoanAmount'].mean()

149.2659574468085

### On average men have loanamounts of 149.26 dollars while average loan amount by females is 126.69

In [13]:
# correlation between loan amount and applicant income
df[['LoanAmount','ApplicantIncome']].corr()

LoanAmount  ApplicantIncome
LoanAmount         1.000000         0.570909
ApplicantIncome    0.570909         1.000000

### There is a moderate relationship between loan amount and applicant income 

In [18]:
df['LoanAmount'].max()

700.0

In [19]:
df[df['LoanAmount']==700]

Loan_ID Gender Married Dependents Education Self_Employed  \
171  LP001585    NaN     Yes         3+  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
171            51763                0.0       700.0             300.0   

     Credit_History Property_Area Loan_Status  
171             1.0         Urban           Y

### The highest loan amount is 700 by a married person has 3 plus dependents, not self employed and the loan is accepted

In [24]:
groupby_status=df.groupby('Loan_Status').count()
groupby_status

Loan_ID  Gender  Married  Dependents  Education  Self_Employed  \
Loan_Status                                                                   
N                192     187      192         186        192            183   
Y                422     414      419         413        422            399   

             ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
Loan_Status                                                                     
N                        192                192         181               186   
Y                        422                422         411               414   

             Credit_History  Property_Area  
Loan_Status                                 
N                       179            192  
Y                       385            422

In [33]:
# The number of married persons and have rejected loans
((df['Married']=='Yes') & (df['Loan_Status']=='N')).sum()

113

In [35]:
# The number of un-married persons and have rejected loans
((df['Married']=='No') & (df['Loan_Status']=='N')).sum()

79

### The Number of rejected loans for married persons was higher than the number of rejected loans for un-married 

In [55]:
df[df['ApplicantIncome']==df['ApplicantIncome'].max()]

Loan_ID Gender Married Dependents Education Self_Employed  \
409  LP002317   Male     Yes         3+  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
409            81000                0.0       360.0             360.0   

     Credit_History Property_Area Loan_Status  
409             0.0         Rural           N

In [ ]:
## The loan is rejected for the person with the highest Income

In [56]:
df[df['ApplicantIncome']==df['ApplicantIncome'].min()]

Loan_ID Gender Married Dependents Education Self_Employed  \
216  LP001722   Male     Yes          0  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
216              150             1800.0       135.0             360.0   

     Credit_History Property_Area Loan_Status  
216             1.0         Rural           N

In [ ]:
## The loan is rejected for the person with the lowest Income

In [57]:
groupby_status=df.groupby('Loan_Status').mean()
groupby_status

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
Loan_Status                                                                     
N                5446.078125        1877.807292  151.220994        344.064516   
Y                5384.068720        1504.516398  144.294404        341.072464   

             Credit_History  
Loan_Status                  
N                  0.541899  
Y                  0.981818

### The average accepted loan amount was 144.29 while the average rejected loan amount was 151 
### There is no relationship between loan_status and applicant income since the average applicant income for rejected loans are higher than income for accepted loans

In [58]:
## Self employeed counts that have accepted loans

In [63]:
((df['Self_Employed']=='Yes') & (df['Loan_Status']=='Y')).sum()

56

In [ ]:
## Self employeed counts that have rejected loans

In [64]:
((df['Self_Employed']=='Yes') & (df['Loan_Status']=='N')).sum()

26

In [65]:
# People Who are self employeed tends to have accepted loans

In [66]:
((df['Gender']=='Male') & (df['Loan_Status']=='N')).sum()

150

In [67]:
((df['Gender']=='Female') & (df['Loan_Status']=='N')).sum()

37

In [68]:
## Males tends to have rejected loans than females

In [76]:
groupby_status=df.groupby('Education').mean()
groupby_status

ApplicantIncome  CoapplicantIncome  LoanAmount  \
Education                                                      
Graduate          5857.433333        1717.474833  154.060215   
Not Graduate      3777.283582        1276.544776  118.409449   

              Loan_Amount_Term  Credit_History  
Education                                       
Graduate            344.669492        0.857788  
Not Graduate        332.156250        0.785124

#### The average of applicant incomes for graduates is higher than the average income for the non graduates 
#### The average loan amount for graduates is higher than the average loan amount for non graduates

In [77]:
groupby_status=df.groupby('Dependents').mean()
groupby_status

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
Dependents                                                                     
0               4917.423188        1623.696000  135.886228        348.107784   
1               5962.274510        1426.235294  158.622449        329.346535   
2               4926.782178        1687.245544  150.224490        340.871287   
3+              8581.215686        2024.313725  190.897959        325.200000   

            Credit_History  
Dependents                  
0                 0.857595  
1                 0.844444  
2                 0.852632  
3+                0.770833

#### The average of income is higher for applicants how have more dependents 
#### The average loan amount got higher for applicants who have larger number of dependents 

## Cross Tabulation tables

In [100]:
df.pivot_table(index='Gender', columns='Loan_Status', aggfunc={'Loan_Status':len})

Loan_Status     
Loan_Status           N    Y
Gender                      
Female               37   75
Male                150  339

In [101]:
df.pivot_table(index='Education', columns='Loan_Status', aggfunc={'Loan_Status':len})

Loan_Status     
Loan_Status            N    Y
Education                    
Graduate             140  340
Not Graduate          52   82

In [103]:
# Percentages is more accurate than number
pd.crosstab(df.Education, df.Loan_Status, normalize='index')

Loan_Status          N         Y
Education                       
Graduate      0.291667  0.708333
Not Graduate  0.388060  0.611940

### 70% of the graduates have accepted loans while 61% of un-graduates have accepted loans

In [106]:
pd.crosstab(df.Dependents, df.Loan_Status, normalize='index')

Loan_Status         N         Y
Dependents                     
0            0.310145  0.689855
1            0.352941  0.647059
2            0.247525  0.752475
3+           0.352941  0.647059

In [107]:
pd.crosstab(df.Gender, df.Loan_Status, normalize='index')

Loan_Status         N         Y
Gender                         
Female       0.330357  0.669643
Male         0.306748  0.693252

### 69% of the Males have accepted loans while 66% of females have accepted loans

In [108]:
pd.crosstab(df.Self_Employed, df.Loan_Status, normalize='index')

Loan_Status           N         Y
Self_Employed                    
No             0.314000  0.686000
Yes            0.317073  0.682927

### The percentage of rejected & accepted loans don't change with the self_employed status